src/steem/serializer.ts

```diff
@@ -99,7 +99,7 @@ const StaticVariantSerializer = (itemSerializers: Serializer[]) => {
  *       Should not be a problem in real-word usage.
  */
 const AssetSerializer = (buffer: ByteBuffer, data: Asset | string | number) => {
-    const asset = Asset.from(data)
+    const asset = Asset.from(String(data))
```

Comment:
Why are you forcing the data into a string here?

Reply:
This was needed to push the sbd_exchange_price: PriceType as part of `set_witness_properties`. Before that, I was getting an error.

Reply:
Hmm, you should never pass a `Price` to the `Asset` serializer

Reply:
AFAIK they were split before into Assets. `sbd_exchange_price` is a PriceType.

Reply:
Casting to string here breaks serialization for other operations. Can you provide an example where setting `sbd_exchange_price` via `set_witness_properties` would fail?

Reply:
Sure


```
let client = new dsteem.Client('https://api.steemit.com')
let base: dsteem.Asset = dsteem.Asset.fromString(`0.800 SBD`)
let quote: dsteem.Asset = dsteem.Asset.fromString(`1.000 STEEM`)

let exchange_rate: dsteem.PriceType = new dsteem.Price(base, quote)

let props: any = { sbd_exchange_rate: exchange_rate }

await essentials.witness_set_properties(client, 'therealwolf', _g.CURRENT_SIGNING_KEY, props, transaction_signing_key)

export let witness_set_properties = async (client: dsteem.Client, owner: string, current_signing_key:string, props: dsteem.utils.WitnessProps, transaction_signing_key: string) => {
  props.key = current_signing_key
  let op = dsteem.utils.buildWitnessUpdateOp(owner, props)
  op[1].props.sort((a, b) => a[0].localeCompare(b[0]))
  return await client.broadcast.sendOperations([op], dsteem.PrivateKey.from(transaction_signing_key))
}
```

Reply:
I tried this and was not able to reproduce any serialisation error

Reply:
Hm, weird. I ran the code myself now (since I cut out the parts for you to test) and it worked as well. So I guess the problem must have been with the way the code was added inside the pricefeed. I've removed the part now and committed it, so it should be ready for merge.